packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -117,34 +117,34 @@ const SentClientRenderFunction /*         */ = 0b00100;
 const SentStyleInsertionFunction /*       */ = 0b01000;
 const SentFormReplayingRuntime /*         */ = 0b10000;
 
-// Per response, global state that is not contextual to the rendering subtree.
-export type ResponseState = {
-  bootstrapChunks: Array<Chunk | PrecomputedChunk>,
+// Per request, global state that is not contextual to the rendering subtree.
+// This cannot be resumed and therefore should only contain things that are
+// temporary working state or are never used in the prerender pass.
+export type RenderState = {
+  // These can be recreated from resumable state.
   placeholderPrefix: PrecomputedChunk,
   segmentPrefix: PrecomputedChunk,
   boundaryPrefix: string,
-  idPrefix: string,
-  nextSuspenseID: number,
-  streamingFormat: StreamingFormat,
 
-  // state for script streaming format, unused if using external runtime / data
+  // inline script streaming format, unused if using external runtime / data
   startInlineScript: PrecomputedChunk,
-  instructions: InstructionState,
 
-  // state for data streaming format
-  externalRuntimeScript: null | ExternalRuntimeScript,
+  // the preamble must always flush before resuming, so all these chunks must
+  // be null or empty when resuming.
 
-  // preamble and postamble chunks and state
+  // preamble chunks
   htmlChunks: null | Array<Chunk | PrecomputedChunk>,
   headChunks: null | Array<Chunk | PrecomputedChunk>,
-  hasBody: boolean,
 
   // Hoistable chunks
   charsetChunks: Array<Chunk | PrecomputedChunk>,
   preconnectChunks: Array<Chunk | PrecomputedChunk>,
   preloadChunks: Array<Chunk | PrecomputedChunk>,
   hoistableChunks: Array<Chunk | PrecomputedChunk>,
 
+  // Module-global-like reference for current boundary resources
+  boundaryResources: ?BoundaryResources,
```

Comment:
It's surprising in a way that BoundaryResources would be render state. maybe worth clarifying here why this is possible?